assets/js/scripts.js

```diff
@@ -0,0 +1,121 @@
+let myLibrary = [];
+
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+}
+
+Book.prototype.toggle = function() {
+  this.read ? (this.read = false) : (this.read = true);
+};
+
+if (localStorage.getItem("book-collection") === null) {
+  const book1 = new Book("First Book", "First Author", 1000, true);
+  const book2 = new Book("Second Book", "Second Author", 344, false);
+  myLibrary.push(book1);
+  myLibrary.push(book2);
+
+  localStorage.setItem("book-collection", JSON.stringify(myLibrary));
+} else {
+  myLibrary = JSON.parse(localStorage.getItem("book-collection"));
+}
+
+const modal = document.getElementById("myModal");
+const btn = document.getElementById("modalBtn");
+const span = document.getElementsByClassName("close")[0];
+
+btn.onclick = function() {
+  modal.style.display = "block";
+};
+
+span.onclick = function() {
+  modal.style.display = "none";
+};
+
+window.onclick = function(event) {
+  if (event.target == modal) {
+    modal.style.display = "none";
+  }
+};
+
+function addBookToLibrary(event) {
+  event.preventDefault();
+  const form = document.querySelector("form");
+  const book = new Book(
+    form.title.value,
+    form.author.value,
+    form.pages.value,
+    form.read.value
+  );
+
+  myLibrary.push(book);
+  localStorage.setItem("book-collection", JSON.stringify(myLibrary));
+
+  modal.style.display = "none";
+
+  //clean form
+  const formFields = document.querySelectorAll("form input");
+  [...formFields].forEach(field => {
+    if (field.getAttribute("type") !== "radio") field.value = "";
+  });
+
+  render();
+}
+
+const addBook = document.getElementById("addBook");
+addBook.addEventListener("click", addBookToLibrary, false);
+
+function removeBookFromLibrary(event) {
+  const cardIndex = event.target.parentNode.getAttribute("data-index");
+  myLibrary = myLibrary.filter((el, i) => i !== +cardIndex);
+  localStorage.setItem("book-collection", JSON.stringify(myLibrary));
+  render();
+}
+
+function render() {
+  const container = document.getElementsByClassName("container")[0];
+  while (container.hasChildNodes()) {
+    container.removeChild(container.firstChild);
+  }
+  myLibrary.forEach((book, index) => {
+    book.__proto__ = Object.create(new Book);
```

Comment:
What did we need this line for guys?

Reply:
Hi Ryan,
The problem is that when we retrieve data from the local storage we get objects literals, so we need to convert them in instances of Book, in order to make the toggle function work.

Reply:
I see. In that case the right way should have been:

```javascript
book.__proto__ = Book.prototype;
```

This converts your ordinary `book` object into an instance of `Book` :)

Reply:
In that case the better way should be:

```javascript
book.__proto__ = Book.prototype
```

Converting your ordinary `book` object into an instance of `Book`

Reply:
Thanks, @rvvergara!